daemon/containerio/container_io.go

```diff
@@ -356,7 +356,16 @@ func (cio *ContainerIO) converge(name, id string, in io.Reader) {
 			break
 		}
 
-		cover, err := cio.ring.Push(data[:n])
+		// FIXME(fuwei): In case that the data slice is reused by the writer,
+		// we should copy the data before we push it into the ringbuffer.
+		// The previous data shares the same address with the coming data.
+		// If we don't copy the data and the previous data isn't consumed by
+		// ringbuf pop action, the incoming data will override the previous data
+		// in the ringbuf.
+		copyData := make([]byte, n)
```

Comment:
From the comment, `Copy` make data not be override, right?

Reply:
Yes

Reply:
More qeustion, as I do not know all ringbuf logic. The data will always append if pop not assume?

Reply:
yes. if the size reaches the capacity, ringbuffer will drop the old one.